File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int maxWater=0;
        int lp=0,rp=height.length-1;

        while(lp<rp){
            //calculate water area
            int width=rp-lp;
            int ht=Math.min(height[lp],height[rp]);
            int currWater=ht*width;
            maxWater=Math.max(maxWater,currWater);
            if(height[lp]<height[rp]){
                lp++;
            }
            else{
                rp--;
            }
        }
        return maxWater;
    }
}

File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        
        m=[]
        i=0
        j=len(height)-1
        while(i<=j):
            if(height[i]<=height[j]):
                h=height[i]
            else:
                h=height[j]
            w=j-i
            m.append(h*w)
            if(height[i]<=height[j]):
                i+=1
            else:
                j-=1
        return max(m)
        
        